micrometer-core/src/main/java/io/micrometer/core/instrument/config/NamingConvention.java

```diff
@@ -73,7 +73,7 @@ public String tagKey(String key) {
         }
 
         private String toCamelCase(String value) {
-            String[] parts = value.split("\\.");
+            String[] parts = value.split("\\.", -1);
```

Comment:
I think there is a behavioral difference here.
`.split("\\.")` is the same as `.split("\\.", 0)` and
>If the limit is zero then the pattern will be applied as many times as possible, the array can have any length, and trailing empty strings will be discarded.
If the limit is negative then the pattern will be applied as many times as possible and the array can have any length.

Though, it seems empty parts will be dropped later on anyways.

Reply:
@izeye What do you think?
I think it would be better to keep the behavior (limit: 0).